tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,615 @@
+        """
+        Method to test the implementation of create_relationship method of class AWSIoT.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = aws.register(test_metric)
+
+        with self.assertRaises(TypeError):
+            # Creating the parent-child relationship between Metric and Edge-System
+            aws.create_relationship(registered_metric, registered_entity)
+
+    def test_validation_create_relationship_parent_entity(self):
+        """
+        Method to test the implementation of create_relationship method of class AWSIoT.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        #  Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = aws.register(test_metric)
+
+        with self.assertRaises(TypeError):
+            # Creating the parent-child relationship between Metric and Edge-System
+            aws.create_relationship(registered_metric, registered_entity)
+
+    def test_validation_create_relationship_child_entity(self):
+        """
+        Method to test the validation of create_relationship method of class AWSIoT.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        with self.assertRaises(TypeError):
+            # Creating the parent-child relationship between Edge-System and Edge-System
+            aws.create_relationship(registered_entity, registered_entity)
+
+    def test_implementation_get_entity_hierarchy(self):
+        """
```

Comment:
Change comment to:
"Test case to check get_entity_entity_hierarchy() for RegisteredEdgeSystem->RegisteredMetric"